tests/test_data_handling.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+import requests
+
+from fobi_custom.plugins.form_handlers.collect_data.fobi_form_handlers import CollectDataPlugin
+from fobi.dynamic import assemble_form_class
+from fobi.models import FormEntry
+
+from pldp.models import Survey, SurveyRow, SurveyComponent
+
+@pytest.mark.django_db
+def test_data_handler(sample_form_entry, sample_location, sample_study, sample_form_element, mocker):
+    request = mocker.MagicMock(spec=requests.Response)
+    form = assemble_form_class(form_entry=sample_form_entry)
+
+    # form.cleaned_data['c75e27cf-4d8f-49dc-bb15-da8137dac247'] = 5
+
+    # A valid CollectDataPlugin requires a DynamicForm, a FormEntry, and a post request
+    plugin = CollectDataPlugin()
+    plugin.run(form=form,
+               form_entry=sample_form_entry,
+               request=request)
+
+    # Check that a survey has been created with a linked row and component
+    survey = Survey.objects.first()
+    row = SurveyRow.objects.filter(survey=survey)
+    component = SurveyComponent.objects.filter(row=row[0])
+
+    assert (len(row), len(component)) == (1, 1)
```

Comment:
I think this is a good assertion to make. But I'd also like you to make some assertions about the content of the newly created entities. Does the SurveyRow and SurveyComponent have the data you expect, given your fixtures?

Reply:
I added a lot more! This now checks for all fields that should be saved. Is this too much? It seems like too much for a single test to handle but I'm not sure how to break it up.